File: app/src/main/java/com/notex/create/EditActivity.java
package com.notex.create;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.Spinner;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.HashMap;

public class EditActivity extends AppCompatActivity {
    
    private EditText titleEdit;
    private EditText contentEdit;
    private Spinner notebookSpinner;
    private SharedPreferences sharedPreferences;
    private Gson gson;
    private ArrayList<HashMap<String, Object>> notebooks;
    private ArrayList<HashMap<String, Object>> notes;
    private String editMode = "new";
    private int editPosition = -1;
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.edit);
        
        // Initialize views
        titleEdit = findViewById(R.id.title_edit);
        contentEdit = findViewById(R.id.content_edit);
        notebookSpinner = findViewById(R.id.notebook_spinner);
        
        // Initialize SharedPreferences and Gson
        sharedPreferences = getSharedPreferences("NoteXData", Context.MODE_PRIVATE);
        gson = new Gson();
        
        // Load data
        loadData();
        
        // Set up spinner
        setupSpinner();
        
        // Check if editing existing note
        if (getIntent().hasExtra("title")) {
            editMode = "edit";
            String title = getIntent().getStringExtra("title");
            
            // Find the note
            for (int i = 0; i < notes.size(); i++) {
                HashMap<String, Object> note = notes.get(i);
                if (note.get("title").toString().equals(title)) {
                    editPosition = i;
                    titleEdit.setText(note.get("title").toString());
                    contentEdit.setText(note.get("content").toString());
                    
                    // Set spinner selection
                    String notebook = note.get("notebook").toString();
                    for (int j = 0; j < notebooks.size(); j++) {
                        if (notebooks.get(j).get("name").toString().equals(notebook)) {
                            notebookSpinner.setSelection(j + 1); // +1 because of "None" option
                            break;
                        }
                    }
                    
                    break;
                }
            }
        }
        
        // Set up back button
        findViewById(R.id.back_button).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
        
        // Set up save button
        findViewById(R.id.save_button).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                saveNote();
            }
        });
    }
    
    private void loadData() {
        // Load notebooks
        String notebooksJson = sharedPreferences.getString("notebooks", "");
        if (!notebooksJson.isEmpty()) {
            Type type = new TypeToken<ArrayList<HashMap<String, Object>>>(){}.getType();
            notebooks = gson.fromJson(notebooksJson, type);
        } else {
            notebooks = new ArrayList<>();
        }
        
        // Load notes
        String notesJson = sharedPreferences.getString("notes", "");
        if (!notesJson.isEmpty()) {
            Type type = new TypeToken<ArrayList<HashMap<String, Object>>>(){}.getType();
            notes = gson.fromJson(notesJson, type);
        } else {
            notes = new ArrayList<>();
        }
    }
    
    private void setupSpinner() {
        ArrayList<String> spinnerItems = new ArrayList<>();
        spinnerItems.add("None");
        
        for (HashMap<String, Object> notebook : notebooks) {
            spinnerItems.add(notebook.get("name").toString());
        }
        
        ArrayAdapter<String> adapter = new ArrayAdapter<>(
            this, android.R.layout.simple_spinner_item, spinnerItems);
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        notebookSpinner.setAdapter(adapter);
    }
    
    private void saveNote() {
        String title = titleEdit.getText().toString().trim();
        String content = contentEdit.getText().toString().trim();
        
        if (title.isEmpty()) {
            Toast.makeText(this, "Please enter a title", Toast.LENGTH_SHORT).show();
            return;
        }
        
        String notebook = "";
        int spinnerPosition = notebookSpinner.getSelectedItemPosition();
        if (spinnerPosition > 0) { // 0 is "None"
            notebook = notebooks.get(spinnerPosition - 1).get("name").toString();
        }
        
        HashMap<String, Object> note = new HashMap<>();
        note.put("title", title);
        note.put("content", content);
        note.put("notebook", notebook);
        
        if (editMode.equals("edit") && editPosition >= 0) {
            notes.set(editPosition, note);
        } else {
            notes.add(note);
        }
        
        // Update notebook counts
        updateNotebookCounts();
        
        // Save notes
        String json = gson.toJson(notes);
        sharedPreferences.edit().putString("notes", json).apply();
        
        // Save notebooks
        json = gson.toJson(notebooks);
        sharedPreferences.edit().putString("notebooks", json).apply();
        
        Toast.makeText(this, "Note saved", Toast.LENGTH_SHORT).show();
        finish();
    }
    
    private void updateNotebookCounts() {
        // Reset all counts
        for (HashMap<String, Object> notebook : notebooks) {
            notebook.put("count", 0);
        }
        
        // Count notes in each notebook
        for (HashMap<String, Object> note : notes) {
            String notebookName = note.get("notebook").toString();
            if (!notebookName.isEmpty()) {
                for (HashMap<String, Object> notebook : notebooks) {
                    if (notebook.get("name").toString().equals(notebookName)) {
                        int count = (int) notebook.get("count");
                        notebook.put("count", count + 1);
                        break;
                    }
                }
            }
        }
    }
}

File: app/src/main/java/com/notex/create/EditActivity.kt
package com.notex.create

import android.os.Bundle
import android.widget.Toast
import androidx.activity.compose.setContent
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Button
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.ExposedDropdownMenuBox
import androidx.compose.material3.ExposedDropdownMenuDefaults
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.material3.TextField
import androidx.compose.material3.TopAppBar
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.livedata.observeAsState
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import com.notex.create.data.Note
import com.notex.create.viewmodel.NoteViewModel

class EditActivity : AppCompatActivity() {

    private val noteViewModel: NoteViewModel by viewModels()
    private var noteId: Int = -1

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        noteId = intent.getIntExtra("id", -1)
        setContent {
            EditScreen(
                noteViewModel = noteViewModel,
                noteId = noteId,
                onSave = {
                    finish()
                }
            )
        }
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun EditScreen(
    noteViewModel: NoteViewModel,
    noteId: Int,
    onSave: () -> Unit
) {
    val context = LocalContext.current
    val notebooks by noteViewModel.allNotebooks.observeAsState(initial = emptyList())
    val notes by noteViewModel.allNotes.observeAsState(initial = emptyList())
    val note = remember(notes, noteId) { notes.find { it.id == noteId } }

    var title by remember { mutableStateOf("") }
    var content by remember { mutableStateOf("") }
    var selectedNotebook by remember { mutableStateOf<Int?>(null) }
    var expanded by remember { mutableStateOf(false) }

    LaunchedEffect(note) {
        if (note != null) {
            title = note.title
            content = note.content
            selectedNotebook = note.notebookId
        }
    }

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text(if (noteId == -1) "New Note" else "Edit Note") }
            )
        }
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .padding(paddingValues)
                .fillMaxSize()
        ) {
            TextField(
                value = title,
                onValueChange = { title = it },
                label = { Text("Title") }
            )
            TextField(
                value = content,
                onValueChange = { content = it },
                label = { Text("Content") }
            )
            ExposedDropdownMenuBox(
                expanded = expanded,
                onExpandedChange = { expanded = !expanded }
            ) {
                TextField(
                    value = notebooks.find { it.id == selectedNotebook }?.name ?: "None",
                    onValueChange = {},
                    readOnly = true,
                    trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(expanded = expanded) },
                    modifier = Modifier.menuAnchor()
                )
                ExposedDropdownMenu(
                    expanded = expanded,
                    onDismissRequest = { expanded = false }
                ) {
                    DropdownMenuItem(
                        text = { Text("None") },
                        onClick = {
                            selectedNotebook = null
                            expanded = false
                        }
                    )
                    notebooks.forEach { notebook ->
                        DropdownMenuItem(
                            text = { Text(notebook.name) },
                            onClick = {
                                selectedNotebook = notebook.id
                                expanded = false
                            }
                        )
                    }
                }
            }
            Button(onClick = {
                if (title.isNotBlank()) {
                    val newNote = Note(
                        id = if (noteId == -1) 0 else noteId,
                        title = title,
                        content = content,
                        notebookId = selectedNotebook
                    )
                    if (noteId == -1) {
                        noteViewModel.insertNote(newNote)
                    } else {
                        noteViewModel.updateNote(newNote)
                    }
                    Toast.makeText(context, "Note saved", Toast.LENGTH_SHORT).show()
                    onSave()
                } else {
                    Toast.makeText(context, "Please enter a title", Toast.LENGTH_SHORT).show()
                }
            }) {
                Text("Save")
            }
        }
    }
}
